src/main/java/io/temporal/internal/statemachines/ActionOrEventType.java

```diff
@@ -0,0 +1,77 @@
+/*
+ *  Copyright (C) 2020 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.internal.statemachines;
+
+import com.google.common.base.Objects;
+import io.temporal.api.enums.v1.CommandType;
+import io.temporal.api.enums.v1.EventType;
+
+class ActionOrEventType<Action> {
```

Comment:
Name of class implies two possible configurations, but it looks like its ActionOrEventTypeOrCommandType (three)

Reply:
[Refactored](https://github.com/temporalio/java-sdk/pull/172/commits/1b1254b636ad9dd6895e8dc2179bede07b2153ae) StateMachine nomenclature to match the standard one.